app/scripts/services/authorization.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("AuthorizationService", function($q, Logger, $interval, DataService){
+    function AuthorizationService() {
+      this.currentProject = null;
+      this.projects = {};
+      this.MAX_PROJECTS = 10;
+    }
+
+    var normalizeRules = function(rules) {
+      var normalizedRules = {};
+      var canAddToProject = null;
+      _.each(rules, function(rule) {
+        _.each(rule.resources, function(resource) {
+          normalizedRules[resource] = {
+            verbs: rule.verbs,
+            apiGroups: rule.apiGroups
+          };
+        });
+        if (_.contains(rule.verbs, "create") || _.contains(rule.verbs, "update")) {
+          canAddToProject = true;
+        }
+      });
+
+      return [normalizedRules, canAddToProject];
+    };
+
+    var getOldestProject = function(allProjectsRules) {
```

Comment:
Sure thing! [Angulars LRU](https://docs.angularjs.org/api/ng/service/$cacheFactory) is pretty easy to use, the `DataService()` constructor fn and  `DataService.prototype._data` are where its used in just a few lines of code.